Improve expander in wxPropertyGrid

MaartenBent opened the pull request:
* Make wxPropertyGrid a wxSystemThemedControl so it uses the default expander
* Don't use hard-coded expander sizes in wxPropertyGrid so the expander gets the correct size, fixes #25571
* Improve margin of sub-groups in wxPropertyGrid so the expander does not overlap with the margin

MaartenBent committed:
Make wxPropertyGrid a wxSystemThemedControl

MaartenBent committed:
Don't use hard-coded expander sizes in wxPropertyGrid

Use wxRendererNative::GetExpanderSize() instead.

Update the tests to work with the bigger expander.

Fixes #25571

MaartenBent committed:
Improve margin of sub-groups in wxPropertyGrid

Make sure the expander of the sub-group does not overlap the margin.

vadz commented:
Unfortunately this doesn't look great under Linux (wxGTK, 200% DPI): this is how it looks in master:

![image](https://github.com/user-attachments/assets/f4579b97-a327-45bb-bcd6-d981eda6f823)

and this is how it looks with this PR: 

![image](https://github.com/user-attachments/assets/510867a6-6ad2-4f90-9637-b46de3bd41ca)

The arrows are clearly scaled, but now they also have too big margins around them. Can this be easily fixed (I didn't look at the code at all to be honest)?

vadz commented:
Sorry, this is not urgent at all but I wanted to ask if you still plan to try something here or should I find some way to reduce the margins myself?

MaartenBent committed:
Reduce wxPropertyGrid gutter width

Using the actual expander icon size instead of hard-coded ones (see 229a9631)
causes the gutter around the icons to be too big.

In general the icon seems wide enough to not need a gutter. Only on Windows
without using the system theme (i.e. when the +/- expander is used) the icon
becomes to small. Calculate a gutter size that will increase the icon width to
a standard 16px width.

MaartenBent commented:
I committed a fix. Seems the gutter is mostly useless now after using the real size of the expander icons.

vadz commented:
Great, thanks a lot! Will merge soon.
